### DISCUSSION TITLE
Scikit-Fuzzy in Micropython

### DISCUSSION THREAD

**Discussion by Nmsngym** [BIPES/BIPES#110] [Q&A] [answered]
Is there a library downloadable or a firmware for scikit-fuzzy in micropython? I will be programming Fuzzy Logic in my RPi Pico. I do not know where to get that module. Pls help.

**Comment by rafaelaroca** [comment]
Hello. Scikit works on BIPES just with Raspberry Pi 1, 2 and 3. Sorry, we did not add support on the pico. Maybe in a near future...
Again, sorry for now.

**Comment by gastmaier** [comment, marked as answer]
Hello,
it is a work in progress. I am/was porting as a part of my term paper, and you can get it here [gastmaier/micropython-scikit-fuzzy](https://github.com/gastmaier/micropython-scikit-fuzzy).
You will need first to compile a custom firmware with [v923z/micropython-ulab](https://github.com/v923z/micropython-ulab), which is a numpy-like array manipulation library for MicroPython (I'm not aware if ulab supports rpi pico).

It is a very complex module, however the plot_cmeans.py and plot_defuzz.py from scikit-fuzzy do work.

I'll provide later a BIPES3 project with these files, so you can upload all of them to your device and try running the examples.

Meanwhile, you can get yourself familiar with the next version of BIPES by:

```
git clone git@github.com:BIPES/BIPES.git
git checkout third
make
```
Ps.1 Only the esp32 toolbox is set and is missing many blocks since I'm porting it.
Ps.2 The intent to port  scikit-fuzzy to MicroPython was testing the limits of the embedded device.

**Reply by rafaelaroca** [reply]
Wow, great!!!

**Reply by Nmsngym** [reply]
Thank you so much!
